app/api/dependencies/product_validator.py

```diff
@@ -0,0 +1,34 @@
+from sqlalchemy.orm import Session
+from app.models import Product
+from app.schemas.product import ProductCreate, ProductUpdate
+from app.api.exceptions.GlobalException import (
+    InvalidProductDataException,
+    ProductAlreadyExistsException,
+)
+
+
+class ProductValidator:
+    def __init__(self, db: Session):
+        self.db = db
+
+    def validate_unique_name(self, name: str):
+        existing_product = self.db.query(Product).filter(Product.name == name).first()
+        if existing_product:
+            raise ProductAlreadyExistsException()
+
+    def validate_product_data(self, product_data: ProductCreate | ProductUpdate):
```

Comment:
I think we don't need this one since we can add it to the base class using paydantic